src/access/AccessRestriction.h

```diff
@@ -0,0 +1,288 @@
+     */
+    static CHIP_ERROR CreateCommissioningEntry(SharedPtr<Entry> entry);
+
+    /**
+     * Add a restriction entry to the access restriction list and notify any listeners. This request
+     * will fail if there is already an entry with the same cluster and endpoint.
+     *
+     * @param [out] index       (If not nullptr) index of created entry (relative to fabric).
+     * @param [in]  entry       The entry to add to the access restriction list.
+     * @param [in]  fabricIndex The index of the fabric in which the entry should be added.
+     * @return                  CHIP_NO_ERROR if the entry was successfully created, or an error code.
+     */
+    CHIP_ERROR CreateEntry(size_t * index, const Entry & entry, const FabricIndex fabricIndex);
+
+    /**
+     * Delete a restriction entry from the access restriction list and notify any listeners.
+     *
+     * @param [in] index           Index of entry to delete.  May be relative to fabric.
+     * @param [in] fabricIndex     Fabric to which entry `index` is relative.
+     * @return                     CHIP_NO_ERROR if the entry was successfully deleted, or an error code.
+     */
+    CHIP_ERROR DeleteEntry(size_t index, const FabricIndex fabricIndex);
+
+    /**
+     * Update a restriction entry in the access restriction list and notify any listeners.
+     *
+     * @param [in] index           Index of entry to delete.  May be relative to fabric.
+     * @param [in] entry        The updated entry to replace the existing entry.
+     * @param [in] fabricIndex  The index of the fabric in which the entry should be updated.
+     * @return                  CHIP_NO_ERROR if the entry was successfully updated, or an error code.
+     */
+    CHIP_ERROR UpdateEntry(size_t index, const Entry & entry, const FabricIndex fabricIndex);
+
+protected:
+    /**
+     * Initiate a review of the access restrictions for a fabric. This method should be implemented by the platform and be
+     * non-blocking.
+     *
+     * @param [in] fabricIndex  The index of the fabric requesting a review.
+     * @param [in] token        The unique token for the review, which can be matched to a review update event.
+     * @param [in] arl          An optinal list of access restriction entries to review.  If null, all entries will be reviewed.
+     * @return                  CHIP_NO_ERROR if the review was successfully requested, or an error code if the request failed.
+     */
+    virtual CHIP_ERROR DoRequestFabricRestrictionReview(const FabricIndex fabricIndex, uint64_t token,
+                                                        const std::vector<Entry> & arl) = 0;
+
+    /**
+     * Clear all access restriction data.
+     */
+    void ClearData();
+
+private:
+    SharedPtr<Entry> GetEntry(FabricIndex fabricIndex, size_t index);
+
+    static std::vector<Platform::SharedPtr<Entry>> sCommissioningEntries;
```

Comment:
This should not be static. This should be a member just like mFabricEntries

Reply:
done.